Merge Puppet module from api-skeleton

michaeldfallen opened the pull request:
This pull request pulls in the puppet module from the [charges-api-skeleton](https://github.com/LandRegistry/charges-api-skeleton).

Commit:
setup for generating unit test code coverage reports

Commit:
Merge pull request #1 from LandRegistry/unit-test-coverage

setup for generating unit test code coverage reports

Commit:
refactored code based on flake8 for best practice

Commit:
refactored helpers file with flake8

Commit:
added puppet files to .gitignore

Commit:
added puppet code

Commit:
fixed metadata bug

michaeldfallen committed:
document the module just in case

michaeldfallen committed:
move module into it's own folder

michaeldfallen committed:
playing around with the class name

michaeldfallen committed:
trying again

Commit:
puppet code added to pull in app

Commit:
added run script to puppet

Commit:
added gunicorn to requirements.txt

michaeldfallen committed:
fix the apps create_manager so that gunicorn can work with it

Commit:
standardised names to _ convention

Commit:
added nginx config

michaeldfallen committed:
Merge pull request #2 from LandRegistry/puppet

Puppet

michaeldfallen committed:
fixed broken test and exposed the need to fix our web hooks

michaeldfallen committed:
small puppet lint fix

michaeldfallen committed:
Merge remote-tracking branch 'skeleton/master'

michaeldfallen committed:
Merge branch 'master' of github.com:LandRegistry/charges-deed-api into merge-skeleton

michaeldfallen committed:
rename the puppet module to reflect the deed_api

endamccormack commented:
Really great work.

Only thing I would say is that the vagrant user is owning all of the files. In my experience its generally best practice to have a user per service.

michaeldfallen committed:
parameterise the owner and group

Pulled in from [charges-borrower-frontend#320a899](https://github.com/LandRegistry/charges-borrower-frontend/commit/320a899b76924e5846b8d2b95a427489676d9a6d).

When running the app in vagrant we would prefer the vagrant user to run
the app, so that when we ssh in we can easily interact with the machine.

When running in the environment we should enable the apps to be run as
their own dedicated user (as old school ops-ers recommend).

To enable this I have parameterised the owner and group and defaulted
them back to vagrant so there is no functional difference.

michaeldfallen committed:
Merge pull request #3 from LandRegistry/parameterise-owner-group

parameterise the owner and group

michaeldfallen committed:
Merge branch 'master' of github.com:LandRegistry/charges-api-skeleton into merge-skeleton

michaeldfallen commented:
Nice catch. I've added that change to the Api Skeleton https://github.com/LandRegistry/charges-api-skeleton/pull/3 and merged the Api Skeleton in to this pull so we get the changes as well.

michaeldfallen committed:
make sure the service runs correctly

michaeldfallen committed:
put the pidfile in a correctly permissioned folder

michaeldfallen committed:
Merge pull request #4 from LandRegistry/run-as-user

Run as user

michaeldfallen committed:
Merge remote-tracking branch 'skeleton/master' into merge-skeleton
